earthfile2llb/converter.go

```diff
@@ -1129,11 +1129,12 @@ func (c *Converter) prepBuildTarget(ctx context.Context, fullTargetName string,
 	return target, opt, propagateBuildArgs, nil
 }
 
-func (c *Converter) buildTarget(ctx context.Context, fullTargetName string, platform *specs.Platform, buildArgs []string, isDangling bool, isFrom bool) (*states.MultiTarget, error) {
+func (c *Converter) buildTarget(ctx context.Context, fullTargetName string, platform *specs.Platform, allowPrivileged bool, buildArgs []string, isDangling bool, isFrom bool) (*states.MultiTarget, error) {
 	target, opt, propagateBuildArgs, err := c.prepBuildTarget(ctx, fullTargetName, platform, buildArgs, isDangling, isFrom)
 	if err != nil {
 		return nil, err
 	}
+	opt.AllowPrivileged = allowPrivileged
```

Comment:
This should go in `prepBuildTarget` instead. This will cover `BuildAsync` too.

Reply:
Good catch, that matches up nicely with the unused allowPriviledged in BuildAsync.